src/common/meta/src/state_store.rs

```diff
@@ -79,19 +80,21 @@ fn decode_kv(kv: KeyValue) -> Result<(String, Vec<u8>)> {
     Ok((key, value))
 }
 
-// Override warnings that those fields are not accessed.
-#[allow(dead_code)]
-enum SplitValue<'a> {
-    Single(&'a [u8]),
-    Multiple(Vec<&'a [u8]>),
+enum SplitValue {
+    Single(Vec<u8>),
+    Multiple(Vec<Vec<u8>>),
 }
 
-fn split_value(value: &[u8], max_value_size: Option<usize>) -> SplitValue<'_> {
+fn split_value(value: Vec<u8>, max_value_size: Option<usize>) -> SplitValue {
     if let Some(max_value_size) = max_value_size {
         if value.len() <= max_value_size {
             SplitValue::Single(value)
         } else {
-            SplitValue::Multiple(value.chunks(max_value_size).collect::<Vec<_>>())
+            let mut values = vec![];
```

Comment:
This also reduce a clone. Since previously:

```rust
    pub fn with_value(mut self, value: impl Into<Vec<u8>>) -> Self {
        self.value = value.into();
        self
    }
```

would clone the inner value chunk.